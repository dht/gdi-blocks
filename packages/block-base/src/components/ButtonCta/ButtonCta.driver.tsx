import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ButtonCta, ButtonCtaProps } from './ButtonCta';
import { BaseComponentDriver } from 'testing-base';

export class ButtonCtaDriver extends BaseComponentDriver {
    private props: Partial<ButtonCtaProps> = {};

    constructor() {
        super('ButtonCta');
    }

    when: any = {
        rendered: () => {
            render(<ButtonCta {...(this.props as ButtonCtaProps)} />);
            return this;
        },
        clicked: () => {
            fireEvent.click(this.wrapper);
            return this;
        },
        snapshot: () => {
            return this.snapshot(<ButtonCta {...(this.props as ButtonCtaProps)} />);
        },
    };

    given: any = {
        props: (props: Partial<ButtonCtaProps>) => {
            this.props = props;
            return this;
        },
    };

    get = {
        WrapperClassName: () => {
            return this.wrapper.className;
        },
        label: () => {
            return this.wrapper.innerHTML;
        },
    };
}
