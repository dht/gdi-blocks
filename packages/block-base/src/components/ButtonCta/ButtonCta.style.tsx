import styled from 'styled-components';

export const Wrapper = styled.div``;

export const Image = styled.img`
    height: 100px;
    box-shadow: 0 0 10px 30px rgba(0, 0, 0, 0.09);
    border-radius: 20px;
`;
