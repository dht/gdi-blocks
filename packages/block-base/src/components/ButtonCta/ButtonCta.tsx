import React from 'react';
import { Image, Wrapper } from './ButtonCta.style';

export type ButtonCtaProps = {};

export function ButtonCta(_props: ButtonCtaProps) {
    return (
        <Wrapper className='ButtonCta-wrapper' data-testid='ButtonCta-wrapper'>
            <Image src='https://static-b9ebe.web.app/button-kickstarter.svg' />
        </Wrapper>
    );
}

export default ButtonCta;
