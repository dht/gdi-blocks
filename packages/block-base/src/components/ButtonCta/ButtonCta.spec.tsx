import { ButtonCtaDriver } from './ButtonCta.driver';
import Chance from 'chance';

const chance = new Chance();

describe('ButtonCta', () => {
    let driver: ButtonCtaDriver;

    beforeAll(() => {
        driver = new ButtonCtaDriver();
    });

    it('should render button', () => {
        const label = chance.word();

        const element = driver.given
            .props({
                title: label,
            })
            .when.rendered();

        const wrapperClassName = element.get.wrapperClassName();
        const innerText = element.get.label();

        expect(wrapperClassName).toContain('ButtonCta-wrapper');
        expect(innerText).toBe(label);
    });

    it('should click button', () => {
        const callback = jest.fn();

        driver.given
            .props({
                onClick: callback,
            })
            .when.rendered()
            .when.clicked();

        expect(callback).toHaveBeenCalledTimes(1);
    });
});

describe('ButtonCta snapshots', () => {
    let driver: ButtonCtaDriver;

    beforeAll(() => {
        driver = new ButtonCtaDriver();
    });

    it('should match snapshot', () => {
        expect(driver.when.snapshot()).toMatchSnapshot();
    });
});
