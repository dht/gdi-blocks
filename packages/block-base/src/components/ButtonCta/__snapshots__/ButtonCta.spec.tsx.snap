// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCta snapshots should match snapshot 1`] = `
<div
  className="sc-bcXHqe ButtonCta-wrapper"
  data-testid="ButtonCta-wrapper"
>
  <img
    className="sc-gswNZR hKPUND"
    src="/button-kickstarter.svg"
  />
</div>
`;
